datafusion/core/src/physical_plan/file_format/parquet/page_filter.rs

```diff
@@ -0,0 +1,514 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Contains code to filter entire pages
+
+use arrow::array::{BooleanArray, Float32Array, Float64Array, Int32Array, Int64Array};
+use arrow::{array::ArrayRef, datatypes::SchemaRef, error::ArrowError};
+use datafusion_common::{Column, DataFusionError, Result};
+use datafusion_expr::utils::expr_to_columns;
+use datafusion_optimizer::utils::split_conjunction;
+use log::{debug, error};
+use parquet::{
+    arrow::arrow_reader::{RowSelection, RowSelector},
+    errors::ParquetError,
+    file::{
+        metadata::{ParquetMetaData, RowGroupMetaData},
+        page_index::index::Index,
+    },
+    format::PageLocation,
+};
+use std::collections::{HashSet, VecDeque};
+use std::sync::Arc;
+
+use crate::physical_optimizer::pruning::{PruningPredicate, PruningStatistics};
+
+use super::metrics::ParquetFileMetrics;
+
+/// Create a RowSelection that may rule out ranges of rows based on
```

Comment:
I polished up some of the existing doc comments to try and explain what this module is doing